src/ODM/MongoDB/Types/CurrencyType.php

```diff
@@ -22,4 +31,9 @@ public function convertToPHPValue($value)
 
         return new Currency($value);
     }
+
+    public function closureToPHP()
+    {
+        return '$return = new \Money\Currency($value);';
```

Comment:
This can be null! Currency can be used outside of money context and could be null.